src/SKVClient/test/SKVRecordTest.cpp

```diff
@@ -0,0 +1,263 @@
+            {dto::FieldType::INT32T, "Balance", false, false}
+    };
+    compatible_schema.setPartitionKeyFieldsByName(std::vector<String>{"LastName"});
+    compatible_schema.setRangeKeyFieldsByName(std::vector<String>{"FirstName"});
+
+    dto::Schema noncompatible_schema;
+    noncompatible_schema.name = "noncompatible_schema";
+    noncompatible_schema.version = 1;
+    noncompatible_schema.fields = std::vector<dto::SchemaField> {
+            {dto::FieldType::INT32T, "LastName", false, false},
+            {dto::FieldType::INT32T, "FirstName", false, false},
+            {dto::FieldType::INT32T, "Balance", false, false}
+    };
+    noncompatible_schema.setPartitionKeyFieldsByName(std::vector<String>{"LastName"});
+    noncompatible_schema.setRangeKeyFieldsByName(std::vector<String>{"FirstName"});
+
+    dto::SKVRecordBuilder builder("collection", std::make_shared<dto::Schema>(schema));
+    builder.serializeNext<String>("a");
+    builder.serializeNext<String>("b");
+    builder.serializeNext<int32_t>(12);
+
+    auto doc = builder.build();
+
+    // Should clone without error
+    dto::SKVRecord clone = doc.cloneToOtherSchema("collection",
+                                    std::make_shared<dto::Schema>(compatible_schema));
+
+    try {
+        // Non-compatible schema, should throw exception
+        dto::SKVRecord clone = doc.cloneToOtherSchema("collection",
+                                    std::make_shared<dto::Schema>(noncompatible_schema));
+        REQUIRE(false);
+    } catch (...) {}
+}
+
+TEST_CASE("Test4: getSKVKeyRecord test") {
+    dto::Schema schema;
+    schema.name = "test_schema";
+    schema.version = 1;
+    schema.fields = std::vector<dto::SchemaField> {
+            {dto::FieldType::STRING, "LastName", false, false},
+            {dto::FieldType::STRING, "FirstName", false, false},
+            {dto::FieldType::INT32T, "Balance", false, false}
+    };
+    schema.setPartitionKeyFieldsByName(std::vector<String>{"LastName"});
+    schema.setRangeKeyFieldsByName(std::vector<String>{"FirstName"});
+
+    dto::SKVRecordBuilder builder("collection", std::make_shared<dto::Schema>(schema));
+    builder.serializeNull();
+    builder.serializeNext<String>("b");
+    builder.serializeNext<int32_t>(12);
+
+    // Testing a typical use-case of getSKVKeyRecord where the storage is serialized
+    // to payload and then read back later
+    dto::SKVRecord key_record = builder.build();
+    const dto::SKVRecord::Storage& storage = key_record.getStorage();
+    MPackWriter w;
+    w.write(storage);
+    Binary buf;
+    REQUIRE(w.flush(buf));
+    MPackReader reader(buf);
+    dto::SKVRecord::Storage read_storage{};
+    REQUIRE(reader.read(read_storage));
+    dto::SKVRecord reconstructed("collection", std::make_shared<dto::Schema>(schema),
+                                     std::move(read_storage), true);
+
+    std::optional<String> last = reconstructed.deserializeNext<String>();
```

Comment:
@iafuture Everything works except this line causes double free error. The stack trace doesn't have symbols. Any suggestion how to debug this issue?